src/contract_wrappers/tokens/security_token_wrapper.ts

```diff
@@ -960,9 +1340,86 @@ export default class SecurityTokenWrapper extends ERC20TokenWrapper {
     );
   };
 
+  public balanceOfByPartition = async (params: BalanceOfByPartitionParams) => {
+    assert.isETHAddressHex('investor', params.tokenHolder);
+    return weiToValue(
+      await (await this.contract).balanceOfByPartition.callAsync(stringToBytes32(params.partition), params.tokenHolder),
+      await this.decimals(),
+    );
+  };
+
+  public transferByPartition = async (params: TransferByPartitionParams) => {
+    assert.isETHAddressHex('To', params.to);
+    assert.assert(params.partition === 'UNLOCKED', 'Invalid Partition');
```

Comment:
Same as above, partitions should be part of an enum

Also, this check should probably be done on a separate function `isValidPartition` that compares the value to all possible partition values. This will make this easier to extend if/when more partitions are added